src/components/side-panel/side-panel-header.tsx

```diff
@@ -29,32 +30,44 @@ export const SidePanelHeader = () => {
     //     if (onSaveChanges) onSaveChanges();
     //     return "Save button clicked";
     // };
-
     const handleCrossButtonClick = () => {
-        toggleMode(EBuilderMode.ADD_ELEMENT);
         removeFocusedElement();
-
+        toggleMode(currentMode);
+        togglePanel(showSidePanel);
```

Comment:
by passing the current states, these toggle functions are not updating anything. they should be removed